autoscale/src/main/java/com/sequenceiq/periscope/monitor/evaluator/load/YarnLoadEvaluator.java

```diff
@@ -129,15 +129,17 @@ protected void pollYarnMetricsAndScaleCluster() throws Exception {
         YarnScalingServiceV1Response yarnResponse = yarnMetricsClient
                 .getYarnMetricsForCluster(cluster, stackV4Response, policyHostGroup, mandatoryDownScaleCount);
 
-        int yarnRecommendedScaleUpCount = yarnResponseUtils.
-                getYarnRecommendedScaleUpCount(yarnResponse, policyHostGroup, maxAllowedUpScale, mandatoryUpScaleCount);
+        int yarnRecommendedScaleUpCount = yarnResponseUtils.getYarnRecommendedScaleUpCount(yarnResponse, policyHostGroup,
+                maxAllowedUpScale, mandatoryUpScaleCount, loadAlertConfiguration.getMaxScaleUpStepSize());
         List<String> yarnRecommendedDecommissionHosts = yarnResponseUtils.
                 getYarnRecommendedDecommissionHostsForHostGroup(cluster.getStackCrn(), yarnResponse,
-                        hostFqdnsToInstanceId, maxAllowedDownScale, mandatoryDownScaleCount);
+                        hostFqdnsToInstanceId, maxAllowedDownScale, mandatoryDownScaleCount, loadAlertConfiguration.getMaxScaleDownStepSize());
 
-        if (yarnRecommendedScaleUpCount > 0) {
+        if (yarnRecommendedScaleUpCount > 0 && isCoolDownTimeElapsed(cluster.getStackCrn(),
+                loadAlertConfiguration.getScaleUpCoolDownMillis(), cluster.getLastScalingActivity()))  {
             sendScaleUpEvent(existingHostGroupSize, yarnRecommendedScaleUpCount);
-        } else if (!yarnRecommendedDecommissionHosts.isEmpty()) {
+        } else if (!yarnRecommendedDecommissionHosts.isEmpty() && isCoolDownTimeElapsed(cluster.getStackCrn(),
+                loadAlertConfiguration.getScaleDownCoolDownMillis(), cluster.getLastScalingActivity()))  {
```

Comment:
Any chance the log message in
EvaluatorExecutor
"LOGGER.debug("Cluster {} cannot be scaled for {} min(s)", clusterCrn," 

could be made a little more specific to differentiate between scale-up/scale-down times for this particular scenario?

Reply:
ok